RFC/src/RFC-0190_Mempool.md

```diff
@@ -0,0 +1,238 @@
+
+- Each [base node] is connected to a number of peers that maintain their own copies of the Mempool.
+
+### Abstract
+
+The Mempool is responsible for managing, verifying and maintaining all unconfirmed transactions that have not yet 
+been included in a [block] and added to the Tari [blockchain]. It consists of a Transaction Pool, Pending Pool, 
+Orphan Pool and Reorg Pool to achieve these tasks. It is also responsible for propagating valid transactions and 
+sharing the Mempool state with connected peers. An overview of the required functionality for the Mempool and each 
+of its component pools will be provided.
+
+### Overview
+
+Every base node maintains a Mempool that consists of four separate pools: the Transaction Pool, Pending Pool, 
+Orphan Pool and Reorg Pool. These four pools have different tasks and work together to form the Mempool used 
+for maintaining unconfirmed transactions. 
+
+This is the role descriptions for each component pool:
+- Transaction Pool: contains all unconfirmed transactions that have been verified, have passed all checks, that 
+only spend valid [UTXO]s and don't have any time-lock restrictions.
+- Pending Pool: contains unconfirmed transactions that have time-lock restrictions. The transactions in this pool 
+either attempt to spend UTXOs with time-locks or the transactions themselves have time-locks limiting them from 
+being included in new blocks until a specified future time or block height has been reached. 
+- Orphan Pool: out of order unconfirmed transactions are managed by this pool, these transactions attempt to spend 
+non-existent UTXOs.
+- Reorg Pool: stores a backup of all transactions that have recently been included into blocks, in case a blockchain 
+reorganization occurs and these transactions have to be restored to the Transaction Pool so that they can be included 
+in future blocks.
+
+### Prioritizing Unconfirmed Transactions
+
+The maximum storage capacity used for storing unconfirmed transactions by the Mempool and each of its component pools 
+can be configured. If a new transaction is received and the storage capacity limits have been reached, then 
+transactions are prioritized according to the transaction priority metric. The transaction priority metric consist 
+of the fees and the maturity of the UTXOs being spent by the transaction and should be applied to determine the priority 
+of each transaction in the Mempool. As the allocated storage space of the Mempool become limited only transactions 
+with the highest priority are kept in the pool. The lowest priority transactions are discarded to make room for 
+higher priority incoming transactions.
+
+The transaction priority metric has the following behavior:
+ - Transactions spending UTXOs with higher block height maturity MUST be prioritized over transactions spending UTXOs 
+ with lower block height maturity.
+ - Transactions with higher fees per transaction message size MUST be prioritized over lower fee transactions.
```

Comment:
Should this be a `SHOULD`? It won't break anything if nodes take the lower fees, though obviously it's in their best interest to do so.

Reply:
Change applied